spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -3,80 +3,117 @@
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
+    end
+
+    # Stripe acepts amounts as positive integers representing how much to charge
+    # in the smallest currency unit
+    let(:capture_amount) { order.total.to_i * 10 }
+    let(:response) { subject.purchase(capture_amount, credit_card, gateway_options) }
+
+    before do
+      # confirms the payment
+      Stripe::PaymentIntent.confirm(payment_intent.id)
+    end
+
+    it "completes the purchase" do
+      payment
 
-      response = subject.purchase(order.total, credit_card, gateway_options)
+      expect {
+        response.to
+        change(
+          Stripe::PaymentIntent.retrieve(payment_intent.id).status
+        ).from("requires_capture").to("suceeded")
+      }
 
       expect(response.success?).to eq true
     end
 
     it "provides an error message to help developer debug" do
-      stub_request(:get, "https://api.stripe.com/v1/payment_intents/12345").
-        to_return(status: 200, body: capture_successful)
+      response_error = subject.purchase(capture_amount, credit_card, gateway_options)
 
-      response = subject.purchase(order.total, credit_card, gateway_options)
+      expect(response_error.success?).to eq false
+      expect(response_error.message).to eq "No pending payments"
+    end
+  end
 
-      expect(response.success?).to eq false
-      expect(response.message).to eq "Invalid payment state: succeeded"
+  context "#error message", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
+
+    let!(:pm_card) do
+      Stripe::PaymentMethod.create({
+                                     type: 'card',
+                                     card: {
+                                       number: '4242424242424242',
+                                       exp_month: 12,
+                                       exp_year: year_valid,
+                                       cvc: '314',
+                                     },
+                                   })
+    end
+    let!(:payment_intent) do
+      Stripe::PaymentIntent.create({
+                                     amount: 1000, # given in AUD cents
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
     end
 
     context "when payment intent state is not in 'requires_capture' state" do
       before do
         payment
       end
 
-      it "succeeds if payment intent state is requires_capture" do
```

Comment:
Just to confirm I'm seeing this right: this success case is now covered above by `it "completes the purchase"`, right?

Reply:
Yes, in the `completes the purchase` case we start off from a payment intent state of "requires_capture" (we check it with `from("requires_capture").to("suceeded")`), because of this, I don't think it makes sense to repeat the test below again.

Reply:
So, this bit was removed. But indeed the state of the payment is `"requires_capture"`, after `Stripe::PaymentIntent.confirm(payment_intent.id)`. So I think we're good with deleting this test case. What do you think?

Reply:
Sounds good to me 👍